rbc/typesystem.py

```diff
@@ -915,25 +914,29 @@ def _normalize(self, target_info):
                 **params)
         raise NotImplementedError(repr(self))
 
-    def mangle(self):
+    def mangle(self, bool_is_int8=None):
         """Return mangled type string.
 
         Mangled type string is a string representation of the type
         that can be used for extending the function name.
         """
+
         if self.is_void:
             return 'v'
         if self.is_pointer:
-            return '_' + self[0].mangle() + 'P'
+            return '_' + self[0].mangle(bool_is_int8=bool_is_int8) + 'P'
         if self.is_struct:
-            return '_' + ''.join(m.mangle() for m in self) + 'K'
+            return '_' + ''.join(m.mangle(bool_is_int8=bool_is_int8)
+                                 for m in self) + 'K'
         if self.is_function:
-            r = self[0].mangle()
-            a = ''.join([a.mangle() for a in self[1]])
+            r = self[0].mangle(bool_is_int8=bool_is_int8)
+            a = ''.join([a.mangle(bool_is_int8=bool_is_int8) for a in self[1]])
             return '_' + r + 'a' + a + 'A'
         if self.is_atomic:
             n = _mangling_map.get(self[0])
             if n is not None:
+                if n == 'b' and bool_is_int8:
```

Comment:
This is problematic as it breaks demangling.

Reply:
Does #143 fixes this?

Reply:
yes

Reply:
So, we shouldn't map `bool` to `int8` anymore? I'm a bit confuse